fslock test failed.

**What happened**:

https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772
```
Cloning into 'gofslock'...
Note: switching to 'cc7f001fe0e7df1710adc8f0cd[9](https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772#step:8:10)e9d6d21fdb3a9'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at cc7f001 Merge pull request #[10](https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772#step:8:11) from fancl20/master
go: downloading golang.org/x/sys v0.0.0-202[11](https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772#step:8:12)216021012-1d35b9e2eb4e
=== RUN   TestConcurrent
=== PAUSE TestConcurrent
=== RUN   TestMultiProcessing
    lock_test.go:186: wrote initial output file to [/jfs/gofslock/fslock/multiprocessing1868176549/out]
    lock_test.go:275: output file doesn't match expected (99 != 100)
--- FAIL: TestMultiProcessing (5.04s)
=== RUN   TestBlockingAndContent
=== PAUSE TestBlockingAndContent
=== RUN   TestUnlock
=== PAUSE TestUnlock
=== RUN   TestSharedConcurrent
=== PAUSE TestSharedConcurrent
=== RUN   TestSharedMultiProcessing
    lock_test.go:587: monitor: acquired exclusive lock
    lock_test.go:596: monitor: observed 'failed_shared' files
    lock_test.go:609: monitor: observed 'has_shared_lock' files
    lock_test.go:6[15](https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772#step:8:16): monitor: failed to re-acquire exclusive lock (0)
    lock_test.go:615: monitor: failed to re-acquire exclusive lock (1)
    lock_test.go:615: monitor: failed to re-acquire exclusive lock (2)
    lock_test.go:615: monitor: failed to re-acquire exclusive lock (3)
    lock_test.go:626: monitor: acquired exclusive lock
--- PASS: TestSharedMultiProcessing (6.09s)
=== RUN   TestExec
=== PAUSE TestExec
=== CONT  TestConcurrent
=== CONT  TestSharedConcurrent
=== CONT  TestExec
=== CONT  TestUnlock
    lock_test.go:431: panicked with: lock is not held
--- PASS: TestUnlock (0.15s)
=== CONT  TestBlockingAndContent
--- PASS: TestSharedConcurrent (0.25s)
--- PASS: TestBlockingAndContent (0.15s)
--- PASS: TestConcurrent (1.02s)
--- PASS: TestExec (11.80s)
FAIL
FAIL	github.com/danjacques/gofslock/fslock	[22](https://github.com/juicedata/juicefs/actions/runs/9196955688/job/25296286772#step:8:23).926s
FAIL
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/vfs/vfs.go
+++ b/pkg/vfs/vfs.go
@@ -633,7 +633,7 @@ func (v *VFS) Release(ctx Context, ino Ino, fh uint64) {
 				v.invalidateAttr(ino)
 			}
 			if locks&1 != 0 {
-				_ = v.Meta.Flock(ctx, ino, fowner, F_UNLCK, false)
+				_ = v.Meta.Flock(ctx, ino, fowner^fh, F_UNLCK, false)
 			}
 			if locks&2 != 0 && powner != 0 {
 				_ = v.Meta.Setlk(ctx, ino, powner, false, F_UNLCK, 0, 0x7FFFFFFFFFFFFFFF, 0)

--- a/pkg/vfs/vfs_unix.go
+++ b/pkg/vfs/vfs_unix.go
@@ -295,7 +295,7 @@ func (v *VFS) Flock(ctx Context, ino Ino, fh uint64, owner uint64, typ uint32, b
 	}
 	h.addOp(ctx)
 	defer h.removeOp(ctx)
-	err = v.Meta.Flock(ctx, ino, owner, typ, block)
+	err = v.Meta.Flock(ctx, ino, owner^fh, typ, block)
 	if err == 0 {
 		h.Lock()
 		if typ == syscall.F_UNLCK {

